os/arch/arm/src/common/up_allocateheap.c

```diff
@@ -129,14 +129,16 @@ void up_allocate_kheap(FAR void **heap_start, size_t *heap_size)
 	/* Return the kernel heap settings (i.e., the part of the heap region
 	 * that was not dedicated to the user heap).
 	 */
-
-	*heap_start = (FAR void *)(g_idle_topstack & ~(0x7));
-
-	/* There may be a special scenario where we might configure a different region
-	 * for heap. In such case, if end of bss falls outside of the region address range,
-	 * then we use the whole region for heap.
-	 */
-	if (*heap_start < (void *)KREGION_START || *heap_start > (void *)KREGION_END) {
+	void *stack_end = (void *)(g_idle_topstack & ~(0x7));
+
+	if ((void *)&_sdata <= (void *)KREGION_END && stack_end >= (void *)KREGION_END) {
+		lldbg("ERROR: Failed to allocate kheap for ram region configuration\n");
+		lldbg("Region start = 0x%x region end = 0x%x\n_sdata = 0x%x end of stack = 0x%x\n",
+				KREGION_START, KREGION_END, &_sdata, stack_end);
```

Comment:
One minor question, 
If we call the PANIC after lldbg, Is that working well?
I mean above is printed well?

Reply:
I am not sure. Because, when there is panic or fault, we have observed that sometimes, the previous logs are missing or getting corrupted. But, some other times, it is printed well. For example, in rtk, the logs are always printed well during crash. But in other boards, we have seen problems sometimes. 
But, what is the alternate method to get the logs?

Reply:
Yeah, that's not the issue in this PR. I just want to know that lldbg is necessary. If it is corrupted always, we don't need to add that.